CHANGELOG/v1.6.0.md

```diff
@@ -0,0 +1,294 @@
+- Release: Add automation to create release branch and tags (#9111)
+- Testing/Documentation: V1.28: Prepare quickstart, capd and tests for the new release including kind bump (#9160)
+
+## :bug: Bug Fixes
+- CAPBK: Certificate paths in cloud-init scripts should not use a platform-dependent path separator (#9167)
+- CAPD: Delete container after failed start to work around port allocation issues (#9125)
+- CAPD: Fix DockerMachine panic (#9673)
+- CI: Fix reporting bug in verify-container-image script (#9676)
+- ClusterClass: Fix ClusterClass enqueue for ExtensionConfig (#9133)
+- ClusterClass: Topology: fix namingstrategy webhook to not use uppercase characters for testing the template and align unit test to e2e test (#9425)
+- clusterctl: Fix provider namespace secret not included in clusterctl move (#9694)
+- ClusterResourceSet: Requeue after 1 minute if ErrClusterLocked got hit (#9788)
+- Dependency: Bump golang.org/x/net to v0.13.0 (#9121)
+- Dependency: Bump to docker v24.0.5-0.20230714235725-36e9e796c6fc (#9038)
+- Devtools: Adding metrics container port in tilt-prepare only if it's missing (#9308)
+- Devtools: Allow duplicate objects in Tiltfile (#9302)
+- Devtools: Change tilt debug base image to golang (#9070)
+- Devtools: Fix tilt-prepare leader-elect setting (#9315)
+- Devtools: Pin Plantuml version (#9424)
+- Devtools: Tilt: ensure .tiltbuild/bin directory is created early enough, add tilt troubleshooting guide (#9165)
+- e2e: Drop MachinePools from Dualstack tests (#9477)
+- e2e: Fix autoscaler image repo (#9353)
+- e2e: Test: pin conformance image to a version which includes a fix for the dualstack tests (#9252)
+- KCP: Allow dropping patches KubeadmControlPlane KubeadmConfig (#9698)
+- KCP: Allow to drop useExperimentalRetryJoin field from KubeadmControlPlane.kubeadmConfigSpec (#9170)
+- KCP: Fix KCP Controller reconcile always return error when workload cluster is unreachable (#9342)
+- KCP: Requeue KCP object if ControlPlaneComponentsHealthyCondition is not yet true (#9032)
+- Machine: Retry Node delete when CCT is locked (#9570)
+- MachineDeployment: Use regular random suffix for MachineSets, ensure max length 63 (#9298)
+- MachineHealthCheck: Fix excessive trace logging in the machine health check controller (#9419)
+- MachinePool: Fix bug where MachinePool Machine ownerRefs weren't updating (#9619)
+- Release: Alphabetically sorting release tool output (#9055)
+- Release: Deduplicating area in pr title in release notes (#9186)
+- Release: Fix go install path for kpromo v4.0.4 (#9336)
+- Release: Generate warning when release notes can not be generated (#9163)
+- Release: Hack/release-notes: ensure relase notes tool can be used for external projects again (#9018)
+- Release: Reverts pull request from cahillsf/improve-release-speed (#9465)
+- Runtime SDK: Set User Agent for test extension correctly (#9748)
+- util: Fix AddAnnotations for unstructured.Unstructured (#9164)
+
+## :seedling: Others
```

Comment:
One more to add: https://github.com/kubernetes-sigs/cluster-api/pull/9798 (or rather the corresponding cherry-pick).

Sorry for the late merge, just makes sense in this case because it's low risk and makes the provider available in clusterctl v1.6.0 directly